velox/functions/sparksql/aggregates/AverageAggregate.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/sparksql/aggregates/AverageAggregate.h"
+#include "velox/functions/lib/aggregates/AverageAggregateBase.h"
+
+#include "velox/functions/lib/aggregates/SumCountChecker.h"
+
+using namespace facebook::velox::functions::aggregate;
+
+namespace facebook::velox::functions::aggregate::sparksql {
+namespace {
+
+template <typename TInput, typename TAccumulator, typename TResult>
+class AverageAggregate
+    : public AverageAggregateBase<TInput, TAccumulator, TResult> {
+ public:
+  explicit AverageAggregate(TypePtr resultType)
+      : AverageAggregateBase<TInput, TAccumulator, TResult>(resultType) {}
+
+  void extractAccumulators(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    auto rowVector = (*result)->as<RowVector>();
+    auto sumVector = rowVector->childAt(0)->asFlatVector<TAccumulator>();
+    auto countVector = rowVector->childAt(1)->asFlatVector<int64_t>();
+
+    rowVector->resize(numGroups);
+    sumVector->resize(numGroups);
+    countVector->resize(numGroups);
+    uint64_t* rawNulls = this->getRawNulls(rowVector);
```

Comment:
Can we just call rowVector->clearAllNulls()?

Reply:
Yes, fixed.